sync/cli/__init__.py

```diff
@@ -52,25 +52,35 @@ def configure():
         default=(CONFIG.default_prediction_preference or Preference.BALANCED).value,
     )
 
-    db_host = click.prompt(
-        "Databricks host (prefix with https://)",
-        default=DB_CONFIG.host if DB_CONFIG else OPTIONAL_DEFAULT,
-    )
-    db_token = click.prompt(
-        "Databricks token",
-        default=DB_CONFIG.token if DB_CONFIG else OPTIONAL_DEFAULT,
-        hide_input=True,
-        show_default=False,
-    )
+    dbx_host = OPTIONAL_DEFAULT
+    dbx_token = OPTIONAL_DEFAULT
+    dbx_region = OPTIONAL_DEFAULT
+    if click.confirm("Would you like to configure a Databricks workspace?"):
+        dbx_host = click.prompt(
+            "Databricks host (prefix with https://)",
+            default=DB_CONFIG.host if DB_CONFIG else OPTIONAL_DEFAULT,
```

Comment:
In playing around with this branch I tried running an existing command,
```
% sync-cli aws-databricks run-job --project load_all_lending_tables 926350091914688
```
which errored out because I didn't have a valid Databricks configuration:
```
% cat ~/.sync/databrickscfg
{
  "host": "https://dbc-d95d06ca-1d00.cloud.databricks.com",
  "token": "derpdedoooo"
}
```
The required region is missing. So I ran `sync-cli configure` which seemed to work, but didn't populate defaults from the existing configuration because it's invalid:
```
% sync-cli configure
Sync API Key ID [SXfC5pbTuHB8FPPy5psUHPgTg7cS9SPm]: 
Sync API Key Secret: 
Default project S3 URL [s3://sync-us-east-1-temp-90-days/scott/projects]: 
Default prediction preference (performance, balanced, economy) [balanced]: 
Would you like to configure a Databricks workspace? [y/N]: y
Databricks host (prefix with https://) [none]:
```
I'd be nice to version these config files to allow for graceful upgrades (I kinda want to get ride of the optional `default_project_url` too). That may not be the right use of resources now though so maybe Pete can weigh in on whether this experience is ok.

Reply:
yeah, I had a similar experience when testing this, and I agree, that would be the best experience! I think the tricky part of this is how to make this new field _required_, but still allow loading the old config that may be missing this newly required field. To allow Pydantic to load it, we'd need to make the model field optional, but the downside is that it's hard to kind of force the user to re-configure to specify this new required data.

Reply:
Maybe we could just default to the AWS region:
```
aws_region_name: str = Field(boto.client("s3").meta.region_name, env="DATABRICKS_AWS_REGION")
```

Reply:
That is an interesting thought 🤔 then we can display the current default in that prompt? I'll give it a try! Good idea